cylc/flow/data_store_mgr.py

```diff
@@ -2367,6 +2367,25 @@ def delta_task_queued(self, itask: TaskProxy) -> None:
         self.state_update_families.add(tproxy.first_parent)
         self.updates_pending = True
 
+    def delta_task_flow_nums(self, itask: TaskProxy) -> None:
```

Comment:
There was no way to update flow_nums in the data_store. 

We got away with this by (erroneously) re-adding flow-merged n=0 tasks to the task pool, which created an entirely new datastore task proxy.